sklearn/metrics/_ranking.py

```diff
@@ -1777,3 +1777,168 @@ def top_k_accuracy_score(
+        + relevance_per_group[tie_group] * (k - counts_before_tie) / counts[tie_group]
+    ) / k
+
+
+def precision_at_k_score(
+    y_true, y_score, *, k=1, sample_weight=None, ignore_ties=False
+):
+    """Compute Precision@K.
+
+    Calculate precision for the top-K scored labels.
+
+    In Information Retrieval paradigm, each sample i represents a query,
+    ``y_true[i]`` - relevance indicators per document (relevant/not relevant),
+    and ``y_score[i]`` - predicted scores per document (used for ranking).
+    The top-scored documents are then considered to be "retrieved"
+    and being evaluated given their true relevance.
+
+    This ranking metric returns a high value if relevant documents are ranked high by
+    ``y_score``. Although the metric takes value in [0, 1] interval,
+    the best scoring function (``y_score``) may not achieve precision@k of 1
+    if the number of positive labels is less than k.
+
+    Parameters
+    ----------
+    y_true : ndarray of shape (n_samples, n_labels)
+        True relevance indicators of entities to be ranked.
+        Any non-zero value is treated as positive/relevant.
+
+    y_score : ndarray of shape (n_samples, n_labels)
+        Target scores, can either be probability estimates, confidence values,
+        or non-thresholded measure of decisions (as returned by
+        "decision_function" on some classifiers).
+
+    k : int, default=1
+        Only consider the highest k scores in the ranking.
+
+    sample_weight : ndarray of shape (n_samples,), default=None
+        Sample weights. If `None`, all samples are given the same weight.
+
+    ignore_ties : bool, default=False
+        Assume that there are no ties in y_score (which is likely to be the
+        case if y_score is continuous) for efficiency gains.
+
+    Returns
+    -------
+    precision_at_k : float in [0., 1.]
+        The averaged precision@k for all samples.
+
+    References
+    ----------
+    `Wikipedia entry for Precision At K
+    <https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)#Precision_at_k>`_
+
+    Manning, Christopher D.; Raghavan, Prabhakar; Schütze, Hinrich (2008).
+    Introduction to Information Retrieval. Cambridge University Press.
+
+    McSherry, F., & Najork, M. (2008, March). Computing information retrieval
+    performance measures efficiently in the presence of tied scores. In
+    European conference on information retrieval (pp. 414-421). Springer,
+    Berlin, Heidelberg.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import precision_at_k_score
+    >>> # we have groud-truth (binary) relevance of some answers to a query:
+    >>> true_relevance = [[0, 1, 0, 1]]
+    >>> # we predict some (relevance) scores for the answers
+    >>> scores = [[0.1, 0.2, 0.3, 0.4]]
+    >>> # we can get the true relevance of the top scored answer (precision@1)
+    >>> precision_at_k_score(true_relevance, scores)
+    1
```

Comment:
Let' me check quickly if that can get the CI green on this PR:

```suggestion
    1.0
```

Reply:
Thank you! :) I had issues running doctest locally 🤦  the suggested way `pytest <module_path>` skips everything, and `python -m doctest <module_path>` fails b/c of local imports. How do you run those tests? 🤔